diff --git a/node_modules/@auth/core/errors.js b/node_modules/@auth/core/errors.js
index 5c580ad..6a4e4e6 100644
--- a/node_modules/@auth/core/errors.js
+++ b/node_modules/@auth/core/errors.js
@@ -139,7 +139,7 @@ InvalidCallbackUrl.type = "InvalidCallbackUrl";
  * 2. If you throw this error in a framework that handles form actions server-side, this error is thrown, instead of redirecting the user, so you'll need to handle.
  */
 export class CredentialsSignin extends SignInError {
-    constructor() {
+    constructor(reason) {
         super(...arguments);
         /**
          * The error code that is set in the `code` query parameter of the redirect URL.
@@ -153,6 +153,7 @@ export class CredentialsSignin extends SignInError {
          * try rather something like "Invalid credentials".
          */
         this.code = "credentials";
+        this.reason = reason;
     }
 }
 CredentialsSignin.type = "CredentialsSignin";
diff --git a/node_modules/@auth/core/index.js b/node_modules/@auth/core/index.js
index 9055748..7248dfb 100644
--- a/node_modules/@auth/core/index.js
+++ b/node_modules/@auth/core/index.js
@@ -130,8 +130,12 @@ export async function Auth(request, config) {
         const isClientSafeErrorType = isClientError(error);
         const type = isClientSafeErrorType ? error.type : "Configuration";
         const params = new URLSearchParams({ error: type });
-        if (error instanceof CredentialsSignin)
+        if (error instanceof CredentialsSignin) {
             params.set("code", error.code);
+            if (error.reason) {
+                params.set("reason", error.reason);
+            }
+        }
         const pageKind = (isAuthError && error.kind) || "error";
         const pagePath = config.pages?.[pageKind] ?? `${config.basePath}/${pageKind.toLowerCase()}`;
         const url = `${internalRequest.url.origin}${pagePath}?${params}`;
diff --git a/node_modules/@auth/core/lib/actions/callback/index.js b/node_modules/@auth/core/lib/actions/callback/index.js
index edcb7f9..6b93b74 100644
--- a/node_modules/@auth/core/lib/actions/callback/index.js
+++ b/node_modules/@auth/core/lib/actions/callback/index.js
@@ -82,6 +82,7 @@ export async function callback(request, options, sessionStore, cookies) {
                     profile: OAuthProfile,
                     isNewUser,
                     trigger: isNewUser ? "signUp" : "signIn",
+                    request,
                 });
                 // Clear cookies if token is null
                 if (token === null) {
@@ -228,12 +229,12 @@ export async function callback(request, options, sessionStore, cookies) {
             const credentials = body ?? {};
             // TODO: Forward the original request as is, instead of reconstructing it
             Object.entries(query ?? {}).forEach(([k, v]) => url.searchParams.set(k, v));
-            const userFromAuthorize = await provider.authorize(credentials, 
+            const userFromAuthorize = await provider.authorize({...credentials, request},
             // prettier-ignore
             new Request(url, { headers, method, body: JSON.stringify(body) }));
             const user = userFromAuthorize;
-            if (!user)
-                throw new CredentialsSignin();
+            if (!user || user?.error)
+                throw new CredentialsSignin(user?.reason || user?.error);
             else
                 user.id = user.id?.toString() ?? crypto.randomUUID();
             const account = {
diff --git a/node_modules/@auth/core/lib/utils/web.js b/node_modules/@auth/core/lib/utils/web.js
index 4e656cf..390f9f7 100644
--- a/node_modules/@auth/core/lib/utils/web.js
+++ b/node_modules/@auth/core/lib/utils/web.js
@@ -23,6 +23,7 @@ export async function toInternalRequest(req, config) {
         const url = new URL(req.url);
         const { action, providerId } = parseActionAndProviderId(url.pathname, config.basePath);
         return {
+            req,
             url,
             action,
             providerId,
diff --git a/node_modules/@auth/core/src/types.ts b/node_modules/@auth/core/src/types.ts
index 36d8a9c..791ae24 100644
--- a/node_modules/@auth/core/src/types.ts
+++ b/node_modules/@auth/core/src/types.ts
@@ -329,6 +329,7 @@ export type AuthAction =
 
 /** @internal */
 export interface RequestInternal {
+  req: Request,
   url: URL
   method: "GET" | "POST"
   cookies?: Partial<Record<string, string>>
